api/proto/common/v2/common_v2.proto

```diff
@@ -20,9 +20,11 @@ message BlobHeader {
   // commitment is the KZG commitment to the blob
   common.BlobCommitment commitment = 3;
   // payment_header contains payment information for the blob
-  common.PaymentHeader payment_header = 4;
+  PaymentHeader payment_header = 4;
   // signature over keccak hash of the blob_header that can be verified by blob_header.account_id
   bytes signature = 5;
+  // salt is used to ensure that the dispersal request is intentionally unique.
+  uint32 salt = 6;
```

Comment:
I don't really like that we're adding stuff after the signature, because the signature signs all of the blob_header, except for itself.

In fact, @litt3 has raised the proposal that we move the signature out of the blob_header, and add is as a separate argument to dispersal requests.

Reply:
Thinking more about this though, in a future where people do self-dispersals and payments is verified by nodes, the signature should actually be part of the blob_header shouldn it?

@ian-shim does the blob_key = blob_header_hash also hash the signature?

Reply:
`blob_key` is packed with `blobHeaderHash` (hashed without signature) and `paymentMetadataHash`, then signer signs over the blob_key to make the signature, which is then written into blobHeader

I'm okay to move salt up but we can discuss if it makes more sense for blob_key to be changed to `(blobHeaderHash, paymentMetadataHash, Salt)`